tools/src/wyvern/target/corewyvernIL/expression/Match.java

```diff
@@ -52,4 +53,14 @@ public Value interpret(EvalContext ctx) {
 		// TODO Auto-generated method stub
 		return null;
 	}
+
+	@Override
+	public Set<String> getFreeVariables() {
+		Set<String> freeVars = matchExpr.getFreeVariables();
+		for (Case c : cases) {
+			freeVars.addAll(c.getBody().getFreeVariables());
```

Comment:
This is trickier.  After getting the free vars of the body of the case, we need to take out the varName bound in the case.  Then we need to add whatever variable is at the root of the path of the NominalType in the case, because we'll look up that variable to get the tag to do the tag check.